app/server/src/routes/user/user.controller.js

```diff
@@ -0,0 +1,152 @@
+        const { email, password } = req.body;
+        try {
+            // Check if user exists in DB
+            const user = await UserModel.getUserByEmail(email);
+            if (!user) {
+                return res
+                    .status(403)
+                    .json({ message: "The user does not exist." });
+            }
+
+            // Hash the req password, compare with the one in db
+            const comparison_result = auth.isPasswordCorrect(user.password_hash, user.password_salt, user.password_iter, password)
+            if (!comparison_result) {
+                return res.status(401).json({
+                    message: "Incorrect Password !",
+                });
+            }
+
+            // If they match, create access token and refresh token, return them 
+            const access_token = await auth.generateToken(email, jwt_ac_secret,access_jwtExpiry)
+            const refresh_token = await auth.generateToken(email, jwt_ref_secret,refresh_jwtExpiry)
+
+            // Save them to DB
+            token_data = {
+                email: email,
+                access_token: access_token,
+                refresh_token: refresh_token
+            };
+            const response = (await UserModel.createToken(token_data));
+            if (response.createdAt) {
+                return res.status(200).json({
+                    id: user._id,
+                    email: email,
+                    access_token: access_token,
+                    refresh_token: refresh_token
+                })
+            } else {
+                return res.status(400).json({
+                    message: "Could not login with the parameters you provided.",
+                });
+            }
+        } catch (error) {
+            return res.status(400).json({
+                message: "Failed to login!",
+                error: error.toString()
+            });
+        }
+    },
+    refresh_tokens: async function (req, res) {
+        const { email, refresh_token } = req.body;// email?
+        try {
+            // does refresh token exist in DB 
+            const tokens = await UserModel.getTokensByEmail(email);
+            if (!tokens) {
+                return res
+                    .status(400)
+                    .json({ message: "The token does not exist." });
+            }
+            // If exists, decrypt
+            jwt.verify(refresh_token, jwt_ref_secret, (err, decoded) => {
+                tokenError = err;
+                decrytedData = decoded;
+                console.log(decoded)
+            });
+            // Acquire email from decrypted token
+            console.log(decrytedData)
+            token_email = decrytedData.email;
+
+            // Token exists, check if it belongs to same user
+            if (email !== token_email) {
+                return res
+                    .status(400)
+                    .json({ message: "The token exists but email mismatch." });
```

Comment:
we shouldn't give the info that the token exists... They could retry with different emails if they have that info that token exists. 
Just returning 400 should be sufficient i believe, or even we could turn 404 user/email not found...

Reply:
Done!